mount-pup: added NTFS3 support

rizalmart opened the pull request:
NTFS3 is new native linux ntfs driver written by paragon software
It will use ntfs3 as primary for mounting ntfs formatted disk if detected otherwise use ntfs-3g as fallback

rizalmart committed:
mount-pup: added NTFS3 support

rizalmart committed:
Update mount-pup

rizalmart committed:
mount-pup: More improvements

01micko reviewed (approved):
Looks ok to me

dimkr commented on woof-code/rootfs-skeleton/bin/mount-pup:
```diff
@@ -86,19 +86,124 @@ elif [ "$NTFS_FLAG" = "yes" ];then
+     
+     [ -f /tmp/ntfs-errors/ntfsmnterr-${MYPID}.txt ] && rm -f /tmp/ntfs-errors/ntfsmnterr-${MYPID}.txt
+      
+      NTFS3_FOUND="$(zcat /proc/config.gz 2>/dev/null | grep "CONFIG_NTFS3_FS=y")"
```
You can just try ntfs3 and fall back to ntfs-3g. It will fail fast, because ntfs3 is an invalid parameter. This long detection sequence will slow down mounting.

rizalmart replied:
Not advisable. ntfs3 might threw exit code number but still mounting ntfs was sucessful. I just remove zcat process to make detection faster.

rizalmart committed:
Update mount-pup

make mount-pup process faster

rizalmart committed:
Use unicode charset on NTFS3

NTFS3 hides filenames contains latin alphabet. So use unicode charset to see those files.

01micko commented:
Sorry, kernel configs aren't ready for this for backward compatibility. Closing.
